Xamarin.Forms.Build.Tasks/CompiledValueProviders/SetterValueProvider.cs

```diff
@@ -43,5 +47,28 @@ public IEnumerable<Instruction> ProvideValue(VariableDefinitionReference vardefr
 			//set the value
 			yield return Instruction.Create(OpCodes.Callvirt, setValueRef);
 		}
+
+		bool IsSetterCollection(FieldReference bindablePropertyReference, ModuleDefinition module, BaseNode node, ILContext context)
```

Comment:
this method doesn't check if this is a Setter at all, I find the name misleading, and I don't understand what this does

Reply:
The idea of the method is to check whether the setter is for a collection; renamed the method to `SetterValueIsCollection` to (hopefully) be more clear. It short-circuits ProvideValue if the Setter is for a collection, since the collection will be intialized later when the first item is added.

Currently I have collections in Setters (e.g. `IList<VisualStateGroup>`) being initialized when the first item is added to them (using `EnsureSetterCollectionInitialized`). I suspect that this should instead be happening in `IValueProvider.ProvideValue`/`ICompiledValueProvider.ProvideValue`, but if there's a simple way to do that, I haven't been able to see it yet. 

Any suggestions?